ush/cdo_post_ocean_format.sh

```diff
@@ -0,0 +1,193 @@
+#
+# Parameters:
+#   $1 - The comma-delimited string to split.
+#
+# Global Variables:
+#   global_array - An array containing the split elements.
+#
+# Example usage:
+#   _comma_split_string "item1,item2 item3,item4"
+#   for element in "${global_array[@]}"; do
+#       echo "$element"
+#   done
+#
+# This example will split the input string into individual elements
+#   and print each element on a separate line.
+function _comma_split_string() {
+    local string="${1}"
+
+    local local_array=()
+    global_array=()
+    IFS="," read -ra items <<< "${string}"
+    for item in "${items[@]}"; do
+        local_array+=("${item} ")
+    done
+    for item in "${local_array[@]}"; do
+        IFS=" " read -ra items <<< "${item}"
+        for element in "${items[@]}"; do
+            global_array+=("${element} ")
+        done
+    done
+}
+
+#######
+
+# _strip_whitespace - Remove whitespace from a string.
+#
+# Description:
+#   This function takes an input string and removes all whitespace
+#   characters (spaces, tabs, and newline characters) to produce a
+#   cleaned output string.
+#
+# Parameters:
+#   $1 - The input string from which whitespace will be removed.
+#
+# Return:
+#   The cleaned string with no whitespace.
+#
+# Example usage:
+#   cleaned_string=$(_strip_whitespace "   This is a string   with spaces  ")
+#   echo "Cleaned string: \"$cleaned_string\""
+#
+# This example will remove all leading, trailing, and internal
+# whitespace from the input string and display the cleaned result.
+function _strip_whitespace(){
+    local in_string="${1}"
+
+    out_string=$(echo "${in_string}" | $(command -v sed) "s/ //g")
+}
+
+#######
+
+# ncattr_update - Update/add attributes for a variable in a netCDF file.
+#
+# Description:
+#   This function updates the specified attribute for a specified
+#     variable in a netCDF file using the `ncatted` command.
+#
+# Parameters:
+#   $1 - The variable name to update.
+#   $2 - netCDF variable metadata attribute name.
+#   $3 - The coordinates as a comma-separated string.
+#
+# Global Variables:
+#   global_array - An array containing the split coordinates.
+#   output_path - The path to the output netCDF file.
+#
+# Example usage:
+#   ncupdate "variable_name" "coords" "lon,lat,time"
+#
+# This example updates the `coords` attributes for the specified
+#   variable and writes the updates to the output netCDF file.
+function ncattr_update(){
+    local varname="${1}"
+    local ncattr="${2}"
+    local coords="${3}"
+
+    _comma_split_string "${coords}"
+    coords="${global_array[@]}"
+    coords_str="$(echo "${coords}" | $(command -v tr) -s ' ')"
+    ncattr_str="$(echo "${ncattr}" | $(command -v tr) -s ' ')"
+    echo "Adding netCDF attribute ${ncattr_str} values ${coords_str} to variable ${varname} metadata and writing to file ${output_path}"
+    ($(command -v ncatted) -O -a "${ncattr_str}","${varname}",c,c," ${coords_str}" "${output_path}" "${output_path}")
```

Comment:
Using `command -v` is overkill.
```suggestion
    (ncatted -O -a "${ncattr_str}","${varname}",c,c," ${coords_str}" "${output_path}" "${output_path}")
```

Reply:
This was introduced for the following reasons and with respect to my understanding of the best-practices for bash scripts.

    Portability: (command -v) is more portable than directly relying on the presence of a specific file or assuming a command is available at a particular location. It works across different Unix-like systems, which may organize their file systems differently.

    Error Handling: It allows you to handle the case where a command is not found gracefully by checking the result and taking appropriate action, such as displaying an error message or exiting the script.

    Flexibility: You can use it in combination with other conditional statements to customize the behavior of your script based on whether a command is available or not.

Reply:
Will leave it up to @aerorahul, but none of our existing scripts use this, and it clutters things up quite a lot once you start using it everywhere.

Reply:
I removed it.